### PR TITLE
Removed Leo from GitOps CODEOWNERS

### PR DESCRIPTION
**Pull request by thschue**

### CODE REVIEW

**Review by AlexsJones — APPROVED** [review]

**Review by scottrigby — CHANGES_REQUESTED** [review]
@thschue Hi, we currently still have all the [maintainers of OpenGitOps](https://github.com/open-gitops/project/blob/main/MAINTAINERS) as codeowners of the `gitops-wg` directory. Leo has stepped down as a co-chair, but is still a maintainer. So we should not merge this PR. Thank you

**Review by AloisReitbauer — APPROVED** [review]

**Review by AloisReitbauer — APPROVED** [review]

**Pull request merged**
